Bosworth Toller's Anglo-Saxon Dictionary Online.js

```diff
@@ -0,0 +1,636 @@
+{
+	"translatorID": "b2d07a2a-c8c6-4426-ba6b-35f094a4d916",
+	"label": "Bosworth Toller's Anglo-Saxon Dictionary Online",
+	"creator": "Zoë C. Ma",
+	"target": "^https://bosworthtoller.com/",
+	"minVersion": "5.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2023-05-03 01:17:45"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2023 Zoë C. Ma
+
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	// The logic is that "a page's type is determined by its content", because
+	// the current implementation of the dictionary web app may not be able to
+	// sync URL correctly all the time.
+	if (getSearchResults(doc, true/* checkOnly */)) {
+		return "multiple";
+	}
+
+	if (doc.querySelector("#btd--entry-single")) {
+		return "dictionaryEntry";
+	}
+
+	// Fallback detection method using URL only.
+	if (url.match(/^https:\/\/bosworthtoller.com\/search(?:\/advanced)?/)) {
+		return "multiple";
+	}
+
+	if (url.match(/^https:\/\/bosworthtoller.com\/\d+/)) {
+		return "dictionaryEntry";
+	}
+
+	return false;
+}
+
+function getSearchResults(doc, checkOnly = false) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll(".btd--search-entry");
+	for (const row of rows) {
+		// Don't retrieve the "similar entry" links.
+		const href = attr(row, ".btd--search-entry-header a", "href");
+		let title = ZU.trimInternal(text(row, ".btd--entry-grammar").trim());
+		if (!title) {
+			title = text(row, ".btd--search-entry-header a"); // fallback
+		}
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found && items;
+}
+
+async function doWeb(doc, url) {
+	if (detectWeb(doc, url) === 'multiple') {
+		const items = await Z.selectItems(getSearchResults(doc));
+		if (!items) return;
+		for (const url of Object.keys(items)) {
+			scrape(await requestDocument(url));
+		}
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+const BOSWORTH_TOLLER_INFO = {
+	dictionaryTitle: "An Anglo-Saxon Dictionary Online",
+	language: "en",
+	place: "Prague",
+	publisher: "Faculty of Arts, Charles University",
+	date: "2014",
+	originalPublisher: "Oxford University Press",
```

Comment:
Are any of these actually saved currently? We can (and should) do this in a citeable way following https://www.zotero.org/support/kb/item_types_and_fields#citing_fields_from_extra